File: project/product/src/main/java/com/codeforce/product/repo/PointRepository.java
package com.codeforce.product.repo;


import com.codeforce.product.model.Point;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Query;
import org.springframework.stereotype.Repository;

import java.util.ArrayList;

@Repository
public interface PointRepository extends JpaRepository<Point, Long> {
    @Query("select p from Point p where p.boxNum <> p.busy")
    ArrayList<Point> findAllFree();


}

File: lesson2/demo/demo/src/main/kotlin/com/example/demo/repo/PointRepository.kt
package com.example.demo.repo

import com.example.demo.model.Point
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.jpa.repository.Query
import org.springframework.stereotype.Repository

@Repository
interface PointRepository: JpaRepository<Point, Long> {

    @Query("select p from Point p where p.boxNum<>p.boxBusy")
    fun findAllFree(): MutableList<Point>?
}